central/graphql/resolvers/image_components.go

```diff
@@ -0,0 +1,115 @@
+	ImageCount(ctx context.Context, args RawQuery) (int32, error)
+	Images(ctx context.Context, args PaginatedQuery) ([]*imageResolver, error)
+	ImageVulnerabilityCount(ctx context.Context, args RawQuery) (int32, error)
+	ImageVulnerabilityCounter(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error)
+	ImageVulnerabilities(ctx context.Context, args PaginatedQuery) ([]ImageVulnerabilityResolver, error)
+	LastScanned(ctx context.Context) (*graphql.Time, error)
+	LayerIndex() *int32
+	License(ctx context.Context) (*licenseResolver, error)
+	Location(ctx context.Context, args RawQuery) (string, error)
+	Name(ctx context.Context) string
+	OperatingSystem(ctx context.Context) string
+	Priority(ctx context.Context) int32
+	RiskScore(ctx context.Context) float64
+	Source(ctx context.Context) string
+	TopImageVulnerability(ctx context.Context) (ImageVulnerabilityResolver, error)
+	UnusedVarSink(ctx context.Context, args RawQuery) *int32
+	Version(ctx context.Context) string
+
+	PlottedVulns(ctx context.Context, args RawQuery) (*PlottedVulnerabilitiesResolver, error) // TODO
+}
+
+// ImageComponent returns an image component based on an input id (name:version)
+func (resolver *Resolver) ImageComponent(ctx context.Context, args IDQuery) (ImageComponentResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Root, "ImageComponent")
+	if !features.PostgresDatastore.Enabled() {
+		return resolver.imageComponentV2(ctx, args)
+	}
+	// TODO : Add postgres support
+	return nil, errors.New("Resolver ImageComponent does not support postgres yet")
+}
+
+// ImageComponents returns image components that match the input query.
+func (resolver *Resolver) ImageComponents(ctx context.Context, q PaginatedQuery) ([]ImageComponentResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Root, "ImageComponents")
+	if !features.PostgresDatastore.Enabled() {
+		query := queryWithImageIDRegexFilter(q.String())
+
+		return resolver.imageComponentsV2(ctx, PaginatedQuery{Query: &query, Pagination: q.Pagination})
+	}
+	// TODO : Add postgres support
+	return nil, errors.New("Resolver ImageComponents does not support postgres yet")
+}
+
+// ImageComponentCount returns count of image components that match the input query
+func (resolver *Resolver) ImageComponentCount(ctx context.Context, args RawQuery) (int32, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Root, "ImageComponentCount")
+	if !features.PostgresDatastore.Enabled() {
+		query := queryWithImageIDRegexFilter(args.String())
+
+		return resolver.componentCountV2(ctx, RawQuery{Query: &query})
+	}
+	// TODO : Add postgres support
+	return 0, errors.New("Resolver ImageComponentCount does not support postgres yet")
+}
+
+func queryWithImageIDRegexFilter(q string) string {
+	return search.AddRawQueriesAsConjunction(q,
+		search.NewQueryBuilder().AddRegexes(search.ImageLabel, ".+").Query())
```

Comment:
Label or ID?

Reply:
There is no `search.ImageId` so I picked a field and made sure it returned data as expected. If there's a more appropriate field I can swap it

Reply:
Ah I see the confusion. Image ID field is named Image Sha to naturally align with the semantics of the field.

Reply:
Sounds good, I'll switch to the usage of sha in that case, thanks